Zarv_challenge_solution

gabriel-delmaestro opened the pull request:
## About You

Name: Gabriel Del Maestro Duarte

Email: gabriel.delmaestro.duarte@gmail.com

- Candidato à vaga de desenvolvedor python júnior, envio, com este pull request, o diretório contendo a minha solução para o desafio proposto

gabriel-delmaestro committed:
Zarv Challenge (solução)

O diretório contém minha solução para o desafio proposto pela empresa Zarv, utilizando da linguagem de programação Python.

gabriel-delmaestro committed:
Atualizando o README

Me esqueci de atualizar o README que havia criado.

gabriel-delmaestro committed:
mudanca README

danisilva02 reviewed (commented):
Olá Gabriel... estou com o seguinte cenário, caso algum usuário selecione um arquivo inválido  ou o mesmo esteja corrompido, como devemos lidar com essa excessão?

cinguilherme commented:
II. A criação da função ETL é de minha responsabilidade, e é passada como parâmetro de antemão (não pode ser alterada em run-time).

Sim a criação é de sua responsabilidade.

gabriel-delmaestro commented:
> Olá Gabriel... estou com o seguinte cenário, caso algum usuário selecione um arquivo inválido ou o mesmo esteja corrompido, como devemos lidar com essa excessão?

Adicionei uma função recursiva (no último commit realizado), chamada check_file, que é encarregada de verificar a validade do arquivo selecionado pelo usuário. Caso o arquivo não seja da extensão .json, ou caso não possa ser lido por algum motivo (talvez corrompido), uma mensagem de erro será exibida no terminal, e uma nova window search box será disponibilizada para seleção de um arquivo compatível.

 Quaisquer outras alterações necessárias, basta me avisar por aqui. Fico à disposição

gabriel-delmaestro commented:
> II. A criação da função ETL é de minha responsabilidade, e é passada como parâmetro de antemão (não pode ser alterada em run-time).
> 
> Sim a criação é de sua responsabilidade.

Perfeito, muito obrigado!

danisilva02 reviewed (commented):
Obrigado pelo retorno Gabriel, vamos analisar a nova inclusão do código que valida os arquivos.

gabriel-delmaestro committed:
update: file check function

Como pedido, inseri uma função recursiva chamada check_file, que vai averiguar se o arquivo escolhido é da extensão .json e se o mesmo pode ou não ser lido. Caso haja algum erro, uma mensagem será exibida no terminal, e uma nova window search box será aberta para que se insira um arquivo válido
